File: semestr-5/Algorytmy i struktury danych 1/bubble_sort.py
def bubble_sort(tab):
    if len(tab) == 0:
        return tab
    swapped = True
    for i in range(len(tab) - 1):
        swapped = False
        for j in range(len(tab) - i - 1):
            if tab[j] > tab[j + 1]:
                tab[j], tab[j + 1] = tab[j + 1], tab[j]
                swapped = True
            print(i,j, tab)
        if not swapped:
            break
    return tab


tab = [11,5,7,2,15,4,6,12,87,36,22,17,91,92,96,97]

bubble_sort(tab)

File: semestr-2/Podstawy programowania/ppr/ANTy_PP/_Pomoce/bubblesort.c
/* Program sortuje tablice nierosnaco algorytmem BubbleSort.
(sortowanie babelkowe) */
#include <stdio.h>
#include <stdlib.h>
#define rozm 20
void zamien(int *a, int *b)
{
 int temp;
temp=*a; *a=*b; *b=temp;
}
void BubbleSort(int tab[], int n)
{
 int i=n-1;
while(i){
if(tab[i]>tab[i-1]){ //wystarczy zmienic znak '<' na '>'
zamien(&tab[i],&tab[i-1]); //zeby otrzymac ciag niemalejacy
i=n-1;
}else
i--;
}
}
void wyp_los(int tab[],int n, int k)
{
 int i; for(i=0;i<n;i++) tab[i]=random(k);
}
void wypisz(int tab[], int n)
{
 int i; for(i=0;i<n;i++) printf("%i, ",tab[i]);
printf("\n");
}
void main()
{
 int t[rozm];
randomize();
wyp_los(t,rozm,rozm);
wypisz(t,rozm);
BubbleSort(t,rozm);
wypisz(t,rozm);
}
